spyder/plugins/ipythonconsole/widgets/debugging.py

```diff
@@ -511,6 +602,21 @@ def _show_prompt(self, prompt=None, html=False, newline=True):
     def _event_filter_console_keypress(self, event):
         """Handle Key_Up/Key_Down while debugging."""
         if self.is_waiting_pdb_input():
+            if (self._reading and self._pdb_single_letter_mode and
+                    event.modifiers() in (Qt.NoModifier, Qt.ShiftModifier)):
+                key = event.text()
+                if len(key) > 0 and key.isprintable():
+                    if key not in (
+                            'a', 'b', 'c', 'd', 'h', 'l',
+                            'n', 'q', 'r', 's', 'u', 'w'):
+                        self._append_plain_text(
+                            "** Error: '" + key + "' is not a pdb command.\n",
+                            before_prompt=True)
+                    else:
+                        self.pdb_execute(self._pdb_cmd_prefix() + key)
+                elif event.key() in (Qt.Key_Escape, Qt.Key_Return, Qt.Key_Enter):
+                    self._toggle_pdb_single_letter_mode()
```

Comment:
This seems handy bur it could cause confusions for our users, i.e. that by pressing these keys (especially Enter) you can switch between single and non-single letter modes.

What if we leave only `Esc` to switch between these modes? That's like a switching between command and edit mode in the notebook, so perhaps people wouldn't be too surprised about it.

Reply:
The problem is that you have a prompt where you only can enter a subset of commands, so it is important to be able to leave it quickly to test something. The use case I have in mind is to use `(n)ext`, `(s)tep` and other "moving" commands until I reach a frame I want to analyse more, at which point I can quickly leave the single letter mode. "Enter" means I want to start typing something, so it is feels quite natural.

Reply:
> The problem is that you have a prompt where you only can enter a subset of commands

Why is that the case? I mean, why is it not possible to enter all Pdb command prefixed by `!`?

Reply:
The idea is that you have the regular mode where you can do whatever you need, and the single letter mode that lets you very quickly explore the code. So instead of typing "!" + "s" + "<Enter>", sou can just type "s" to step through the code. When you reach a frame you want to analyse, you want to exit quickly. Because you r finger is likely already above the "Enter" key because you enter a lot of short commands, it makes sense to be able ot use it to enter the regular mode whee it is useful.

Reply:
> Because you r finger is likely already above the "Enter" key because you enter a lot of short commands, it makes sense to be able ot use it to enter the regular mode whee it is useful.

I understand your point, but since `Enter` is primarily used for execution, associating this additional functionality to it could be very confusing. That's why I think we should only use `Esc` to switch between these modes.

Reply:
I moved that to a setting that is disabled by default if that is ok?

Reply:
Yes, that's ok for me. Thanks for doing it!